### PR TITLE
Fix/cli new acc

### PR DESCRIPTION
**Pull request by l4l**
### Description of the Change

- Add separation of KeysManager's create/load methods that work with and without pass phrase
- Key is randomly generated
- Pass phrase now is used for encryption/decryption (for now pretty dummy xor) of the private key on the disk (can be empty, then stored as plain text)
- Add some related tests

### Benefits

No guessable keys, more clear iface for KeysManager

### Possible Drawbacks 

Key can be lost and there's no way of reverting it back

### Usage Examples or Tests *[optional]*

```bash
# saves plain text keys
iroha-cli --new_account --name test@acc

# saves plain text public key and encrypted private key
iroha-cli --new_account --name test@acc --pass_phrase 12345
```
### TODO so far

- Create key file structure
- Replace encryption scheme

### CODE REVIEW

File: libs/crypto/keys_manager_impl.cpp [hunk 1]
```diff
@@ -36,23 +38,53 @@ namespace iroha {
...
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
+      std::string key(binstr.size(), '\0');
+      const auto pass_size = std::max(1ul, pass_phrase.size());
+
+      for (auto i = 0u; i < binstr.size(); i++) {
+        key[i] = binstr[i] ^ pass_phrase[i % pass_size];
+      }
+      return nonstd::make_optional(key);
```
Comment (kamilsa) [inline libs/crypto/keys_manager_impl.cpp]: why return optional here?

File: libs/crypto/keys_manager.hpp [hunk 2]
```diff
@@ -38,12 +38,27 @@ namespace iroha {
...
+     * keypair by signing and verifying signature of test message
+     * @param pass_phrase is the key for decryption
+     * @return nullopt if no keypair found locally, or verification failure
+     */
+    virtual nonstd::optional<iroha::keypair_t> loadKeys(
+        const std::string &pass_phrase) = 0;
+
+    /**
+     * Create a new keypair and store it as is on disk
+     * @param pass_phrase is password for the keys
```
Comment (kamilsa) [inline libs/crypto/keys_manager.hpp]: No such param here

File: libs/crypto/keys_manager_impl.cpp (hunk 1 above)
Comment (l4l) [reply libs/crypto/keys_manager_impl.cpp]: For binding here:
https://github.com/hyperledger/iroha/blob/38862e33674375535eca98168c3a951e57ecdb10/libs/crypto/keys_manager_impl.cpp#L131-L134

File: libs/crypto/keys_manager_impl.cpp [hunk 3]
```diff
@@ -20,39 +20,85 @@
...
+   * key via XORing with pass phrase
+   * @param s is an encrypted data from file
+   * @param pass_phrase for decryption
+   * @return function that will set keypair::privkey on successful
+   *         deserialization and decryption
+   */
+  auto deserializedEncrypted(const std::string &s,
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
```
Comment (luckychess) [inline libs/crypto/keys_manager_impl.cpp]: May be move decryption into separate function, like encrypt?

File: libs/crypto/keys_manager_impl.cpp [hunk 4]
```diff
@@ -20,39 +20,85 @@
...
         log_(logger::log("KeysManagerImpl")) {}
 
-  bool KeysManagerImpl::validate(const iroha::keypair_t &keypair) const {
+  bool KeysManagerImpl::validate(const keypair_t &keypair) const {
     std::string test = "12345";
-    auto sig = iroha::sign(
-        iroha::sha3_256(test).to_string(), keypair.pubkey, keypair.privkey);
-    if (not iroha::verify(
-            iroha::sha3_256(test).to_string(), keypair.pubkey, sig)) {
+    auto sig =
```
Comment (luckychess) [inline libs/crypto/keys_manager_impl.cpp]: Looks like randomString() from generator.cpp can be used.

File: libs/crypto/keys_manager_impl.cpp [hunk 5]
```diff
@@ -20,39 +20,85 @@
...
+   */
+  auto deserializedEncrypted(const std::string &s,
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
+      std::string key(binstr.size(), '\0');
+      const auto pass_size = std::max(1ul, pass_phrase.size());
+
+      for (auto i = 0u; i < binstr.size(); i++) {
+        key[i] = binstr[i] ^ pass_phrase[i % pass_size];
```
Comment (luckychess) [inline libs/crypto/keys_manager_impl.cpp]: Please add TODO with task id in Jira for replacing XOR encryption with something harder to brake.

**Comment by bakhtin** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 7 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [main.cpp#L52](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L52): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AHardcodedIp)
1. ![MINOR][MINOR] [main.cpp#L40](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L40): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L42](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L42): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L43](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L43): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L45](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L45): Define a constant instead of duplicating this literal false 4 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L46](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L46): Define a constant instead of duplicating this literal "Name of the account" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L54](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L54): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L101](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/iroha-cli/main.cpp#L101): Define a constant instead of duplicating this literal "Json has wrong format." 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L46](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/irohad/model/generators/impl/transaction_generator.cpp#L46): Define a constant instead of duplicating this literal "admin" 3 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L48](https://github.com/hyperledger/iroha/blob/ccf3171aefd5fa71dcd3e299d4979b3b9b753d8a/irohad/model/generators/impl/transaction_generator.cpp#L48): Define a constant instead of duplicating this literal "user" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: libs/crypto/keys_manager_impl.cpp (hunk 5 above)
Comment (l4l) [reply libs/crypto/keys_manager_impl.cpp]: [Here](https://soramitsu.atlassian.net/browse/IR-863) it is, don't really sure that it should be added as a todo

File: libs/crypto/keys_manager_impl.cpp (hunk 4 above)
Comment (l4l) [reply libs/crypto/keys_manager_impl.cpp]: Quite out of that pr, anyway that's easy to fix

**Comment by bakhtin** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 7 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [main.cpp#L52](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L52): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AHardcodedIp)
1. ![MINOR][MINOR] [main.cpp#L40](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L40): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L42](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L42): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L43](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L43): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L45](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L45): Define a constant instead of duplicating this literal false 4 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L46](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L46): Define a constant instead of duplicating this literal "Name of the account" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L54](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L54): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L101](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/iroha-cli/main.cpp#L101): Define a constant instead of duplicating this literal "Json has wrong format." 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L46](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/irohad/model/generators/impl/transaction_generator.cpp#L46): Define a constant instead of duplicating this literal "admin" 3 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L48](https://github.com/hyperledger/iroha/blob/98af97383ce9414a11b6c3cb0f345b4fe5325a62/irohad/model/generators/impl/transaction_generator.cpp#L48): Define a constant instead of duplicating this literal "user" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by bakhtin** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 7 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [main.cpp#L48](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L48): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AHardcodedIp)
1. ![MINOR][MINOR] [main.cpp#L36](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L36): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L38](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L38): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L39](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L39): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L41](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L41): Define a constant instead of duplicating this literal false 4 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L42](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L42): Define a constant instead of duplicating this literal "Name of the account" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [main.cpp#L50](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L50): Remove this commented out code. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3ACommentedCode)
1. ![MINOR][MINOR] [main.cpp#L97](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/iroha-cli/main.cpp#L97): Define a constant instead of duplicating this literal "Json has wrong format." 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L46](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/irohad/model/generators/impl/transaction_generator.cpp#L46): Define a constant instead of duplicating this literal "admin" 3 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)
1. ![MINOR][MINOR] [transaction_generator.cpp#L48](https://github.com/hyperledger/iroha/blob/5894a41510f32176e6f38a3cd7f5c09176bfb9fc/irohad/model/generators/impl/transaction_generator.cpp#L48): Define a constant instead of duplicating this literal "user" 2 times. [rule](https://sonar.soramitsu.co.jp/coding_rules#rule_key=cxx%3AStringLiteralDuplicated)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**